### ISSUE TITLE
HealthChecks.CosmosDB - Deferred CosmosClient resolution

### ISSUE THREAD

**Issue by jwyza-pi** [Xabaril/AspNetCore.Diagnostics.HealthChecks#1567]
**What would you like to be added:**

I'd like to a factory method that allows for resolution of the `CosmosClient` to be controlled similar to how the configureOptions is able to be resolved with the `IServiceProvider`. 

```csharp
        /// <summary>
        /// Add a health check for Azure CosmosDb database.
        /// </summary>
        /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
        /// <param name="clientFactory">Delegate for creating the <see cref="CosmosClient"/> instance that will communicate with the database.</param>
        /// <param name="configureOptions">Delegate for configuring the health check. Optional.</param>
        /// <param name="name">The health check name. Optional. If <see langword="null"/> the type name 'cosmosdb' will be used for the name.</param>
        /// <param name="failureStatus">
        /// The <see cref="HealthStatus"/> that should be reported when the health check fails. Optional. If <see langword="null"/> then
        /// the default status of <see cref="HealthStatus.Unhealthy"/> will be reported.
        /// </param>
        /// <param name="tags">A list of tags that can be used to filter sets of health checks. Optional.</param>
        /// <param name="timeout">An optional <see cref="TimeSpan"/> representing the timeout of the check.</param>
        /// <returns>The specified <paramref name="builder"/>.</returns>
        public static IHealthChecksBuilder AddCosmosDb(
            this IHealthChecksBuilder builder,
            Func<IServiceProvider, CosmosClient> clientFactory,
            Action<IServiceProvider, CosmosDbHealthCheckOptions>? configureOptions = default,
            string? name = default,
            HealthStatus? failureStatus = default,
            IEnumerable<string>? tags = default,
            TimeSpan? timeout = default)
        {
            return builder.Add(new HealthCheckRegistration(
               name ?? COSMOS_NAME,
               sp =>
               {
                   var options = new CosmosDbHealthCheckOptions();
                   configureOptions?.Invoke(sp, options);
                   return new CosmosDbHealthCheck(clientFactory(sp), options);
               },
               failureStatus,
               tags,
               timeout));
        }

```

**Why is this needed:**

Our CosmosClients aren't always initialized as soon as the application starts up and we will have more than one because we have multiple cosmos DBs (so 1 client per DB, not per container).  We have a factory that lazy initializes them the first time they are needed.  Additionally the configuration for these clients is injected via DI initially.  As such we need to be able to defer resolution of the client.  By adding the ability to return the `CosmosClient` ourselves during runtime, we can ensure that the client we want is returned.

I'd be happy to go through the steps to get this in if it makes sense.

PS - Sorry for the 2nd issue on this.  I had made one for the documentdb package, but realized that was the legacy SDK.  I had hoped that this new package would work as it had deferred resolution of some things, but after digging through the source, it turned out it wasn't quite able to do what I needed.

**Comment by sungam3r** [comment]
PR is welcome.

**Comment by sungam3r** [comment]
CosmosDbHealthCheckBuilderExtensions.AddAzureCosmosDB
